doc/start/quick-cephfs.rst

```diff
@@ -3,117 +3,210 @@
+appropriate value for ``pg_num``, refer `Placement Group`_. You can also use
+pg_autoscaler plugin instead. Introduced by Nautilus release, it can
+automatically increase/decrease value of ``pg_num``; refer the
+`Placement Group`_ to find out more about it.
 
-	cat ceph.client.admin.keyring
+When all OSDs are on the same node...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+And, in case you have deployed all of the OSDs on the same node, you would need
+to create a new CRUSH rule to replicate data across OSDs and set the rule on the
+CephFS pools, since the default CRUSH rule is to replicate data across
+different nodes::
 
-#. Copy the key of the user who will be using the mounted CephFS file system.
-   It should look something like this:: 
-	
-	[client.admin]
-	   key = AQCj2YpRiAe6CxAA7/ETt7Hcl9IyxyYciVs47w==
+    ceph osd crush rule create-replicated rule_foo default osd
+    ceph osd pool set cephfs_data crush_rule rule_foo
+    ceph osd pool set cephfs_meta crush_rule rule_foo
 
-#. Open a text editor. 
+Using Erasure Coded pools
+^^^^^^^^^^^^^^^^^^^^^^^^^
+You may also use Erasure Coded pools which can be more effecient and
+cost-saving since they allow stripping object data across OSDs and
+replicating these stripes with encoded redundancy information. The number
+of OSDs across which the data is stripped is `k` and number of replica is `m`.
+You'll need to pick up these values before creating CephFS pools. The
+following commands create a erasure code profile, creates a pool that'll
+use it and then enables it on the pool::
 
-#. Paste the key into an empty file. It should look something like this::
+    ceph osd erasure-code-profile set ec-42-profile k=4 m=2 crush-failure-domain=host crush-device-class=ssd
+    ceph osd pool create cephfs_data_ec42 64 erasure ec-42-profile
+    ceph osd pool set cephfs_data_ec42 allow_ec_overwrites true
+    ceph fs add_data_pool mycephfs cephfs_data_ec42
 
-	AQCj2YpRiAe6CxAA7/ETt7Hcl9IyxyYciVs47w==
+You can also mark directories so that they are only stored on certain pools::
 
-#. Save the file with the user ``name`` as an attribute 
-   (e.g., ``admin.secret``).
+    setfattr -n ceph.dir.layout -v pool=cephfs_data_ec42 /mnt/mycephfs/logs
 
-#. Ensure the file permissions are appropriate for the user, but not
-   visible to other users. 
+This way you can choose the replication strategy for each directory on your
+Ceph file system.
 
+.. note:: Erasure Coded pools can not be used for CephFS metadata pools.
 
-Kernel Driver
-=============
+Erasure coded pool were introduced in Firefly and could be used directly by
+CephFS Luminous onwards. Refer `this article <https://ceph.io/community/new-luminous-erasure-coding-rbd-cephfs/>`_
+by Sage Weil to understand EC, it's background, limitations and other details
+in Ceph's context. Read more about `Erasure Code`_ here.
+
+Mounting the File System
+========================
+
+Using Kernel Driver
+-------------------
```

Comment:
Please change the header style here, make it bit bold.

Reply:
True. But I've followed the pattern in [dev guide doc](https://docs.ceph.com/docs/master/dev/developer_guide/), titles underlined by "=", are followed by titles followed by "-" and then title underlined by "^". Something's wrong with the rendering I suppose,